### PR TITLE
Add hashids.1.0.0 and dependencies

### PR DESCRIPTION
**Pull request by jacquev6**
[hashids](http://hashids.org/): generate short, unique, non-sequential ids from numbers, that you can also decode

[Homepage](https://jacquev6.github.io/hashids-ocaml/)

### CODE REVIEW

File: packages/General/General.0.3.0/opam [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    "_build/src/General.a"
+    "_build/src/General.cmxa"
+    "_build/src/Foundations.cmx"
+    "_build/src/Traits.cmx"
+    "_build/src/Concepts.cmx"
+    "_build/src/Implementation.cmx"
+    "_build/src/Testing.cmx"
+    "_build/src/Specializations.cmx"
+    "_build/src/General.cmx"
+    "%{prefix}%/lib/General"
```
Comment (samoht) [inline packages/General/General.0.3.0/opam]: This is a rather crude way to install the package :-) What happens if there is no native compiler?  I am fine to merge this for an intermediate version, but this is really fragile.

File: packages/hashids/hashids.1.0.0/opam [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  ["mkdir" "-p" "%{prefix}%/lib/hashids"]
+  [
+    "cp"
+    "META"
+    "_build/Hashids.cmi"
+    "_build/Hashids.cma"
+    "_build/Hashids.a"
+    "_build/Hashids.cmx"
+    "_build/Hashids.cmxa"
+    "%{prefix}%/lib/hashids"
```
Comment (samoht) [inline packages/hashids/hashids.1.0.0/opam]: same thing here: this is very fragile and will fail if you don't have a native compiler installer (not sure what will happen on windows). Not sure if it is a very good to merge this for a 1.0 version, but this is up to you.

**Comment by jacquev6** [comment]
@samoht Thanks for your comments! Would you be able to point me at a *simple* example of a proper way to do this?

I was thinking that copying the ".cmi" and ".cma" files was enough for a bytecode compiler, and that the ".cmx" ".a" and ".cmxa" files were for the native compiler. Maybe we're not talking about the same thing. Anyway I'm willing to learn!

**Comment by samoht** [comment]
Since a few months I've been using `jbuilder` and I completely forgot about all of this stuff :-) 

More seriously I don't remember if pack files contain all the information for cross-module inlining and other linking information (probably yes). You are also missing the `cmxs` needed for dynamic linking. Also in the case the native compiler is not available the `cmx` files will not be produced and the `cp` will fail as it won't be able to find them. There are a lot of "interesting" cases that higher-level description tools (such as [topkg](http://erratique.ch/software/topkg/doc/Topkg.Pkg.html#VALmllib) or [jbuilder](https://github.com/janestreet/jbuilder/blob/master/src/gen_rules.ml#L854-L926) are solving pretty well.

I had a quick look at your libraries and it should be easy enough to port to `jbuilder` apart the use of bisect (see [here](https://github.com/janestreet/jbuilder/issues/57) for some context).

**Comment by jacquev6** [comment]
OK, thanks! I'll have a look at jbuilder. I now realize @avsm was saying the same thing in #10147.

Don't merge this PR, I'll open a new one when I'm done.

**Pull request closed without merging**